main.tf

```diff
@@ -374,14 +381,20 @@ echo "#TSMCMD accept EULA - only required for tableau_srv"
 su -c "tabcmd --accepteula" - tableau_srv
 
 echo "#TSMCMD - initial user"
-tabcmd initialuser --server 'localhost:80' --username $TAB_ADMIN_USER --password $TAB_ADMIN_PASSWORD
+tabcmd initialuser --server 'localhost:80' --username "$TAB_ADMIN_USER" --password "$TAB_ADMIN_PASSWORD"
 
 echo "#Get most recent Tableau backup from S3"
 export LATEST_BACKUP_NAME=`aws s3 ls $DATA_ARCHIVE_TAB_BACKUP_URL | tail -1 | awk '{print $4}'`
 aws s3 cp $DATA_ARCHIVE_TAB_BACKUP_URL$LATEST_BACKUP_NAME /var/opt/tableau/tableau_server/data/tabsvc/files/backups/$LATEST_BACKUP_NAME
 
 echo "#Restore latest backup to Tableau Server"
-tsm stop -u $TAB_SRV_USER -p $TAB_SRV_PASSWORD && tsm maintenance restore --file $LATEST_BACKUP_NAME -u $TAB_SRV_USER -p $TAB_SRV_PASSWORD && tsm start -u $TAB_SRV_USER -p $TAB_SRV_PASSWORD
+tsm stop -u "$TAB_SRV_USER" -p "$TAB_SRV_PASSWORD" && tsm maintenance restore --file $LATEST_BACKUP_NAME -u "$TAB_SRV_USER" -p "$TAB_SRV_PASSWORD" && tsm start -u "$TAB_SRV_USER" -p "$TAB_SRV_PASSWORD"
+
+###Publish the *required* workbook(s)/DataSource(s) - specified somehow...?
+/home/tableau_srv/tableau-login.sh DQDashboards
+/home/tableau_srv/tableau-pub-ds.sh "Data Receipt" "/home/tableau_srv/tableau-dq/datasources/Data Receipt/Cv_Exploratory_Inbound_SP.tdsx"
```

Comment:
Can these be passed in from SSM rather than hardcoding in `user_data`? Is `tableau_int_publish_datasources` not used for `$DATASOURCE_FILENAME`?
Also it seems `$PROJECT` setup is missing from SSM/bash export

Reply:
Ultimately, yes, the list of which DataSources and WorkBooks to publish will be supplied in Param Store variables, i.e. tableau_int_publish_datasources for `DataSources` and eventually `tableau_int_publish_workbooks` for WorkBooks. However, for now, as I've made so many changes I want to see what works and what breaks (is there a way to do this without running into NotProd with Drone?). I also need to see what format the SSM "String List" will appear as in User Data (a quick search of the web did not provide me with the answer).

`PROJECT` will be specified as part of the DataSource/WorkBook to be published.